language/Type-Classes.md

```diff
@@ -106,6 +106,10 @@ instance semigroupAddInt :: Semigroup AddInt where
 
 In fact, a type similar to this `AddInt` is provided in `Data.Monoid.Additive`, in the `monoid` package.
 
+The reason why orphan instances are banned is because they can lead to two types of problems:
+* Recall that only one instance is allowed per type and class combination in a module. Orphan instances enable duplicate instances to be defined in separate modules without breaking this overlapping instances rule. So these modules are fine independently, but if both modules are ever imported into the same project (for example from separate libraries), then an instance collision would break the build.
```

Comment:
This bullet point moves back and forth between the way things actually are (i.e. orphans are banned) and the way things might be if they weren't, and so I think it could be made a bit tighter by addressing that. Additionally, I think a slightly more explicit example would help. How about:

> Suppose two separate modules define an orphan `Semigroup Int` instance, and one of them uses `+` for `append`, whereas the other uses `*`. Now suppose someone writes a third module which imports both of the first two, and that somewhere in that third module we have the expression `2 <> 3`, which calls for a `Semigroup Int` instance. The compiler now has two instances to choose from. What should it do? It could report an error, or it could arbitrarily pick one of the instances. Neither option is particularly appealing:
>
> * If it chooses to report an error, it means that any pair of modules which define the same orphan instance can never be used together.
> * If it arbitrarily picks one, we won't be able to determine whether `2 <> 3` will evaluate to 5 or 6. This can make it very difficult to ensure that your program will behave correctly!

Reply:
That example ties in nicely with the previous code snippet. Included it in the latest push.

Reply:
Thanks! That's why I used it :slightly_smiling_face: